### PR TITLE
[Windows] GH Actions

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Omit
Docs: Omit
Related to https://github.com/conan-io/conan/pull/17094 https://github.com/conan-io/conan/pull/17054

### CODE REVIEW

File: conan/internal/default_settings.py [hunk 1]
```diff
@@ -111,7 +111,7 @@
         cstd: [null, 99, gnu99, 11, gnu11, 17, gnu17, 23, gnu23]
     msvc:
         version: [170, 180, 190, 191, 192, 193, 194]
-        update: [null, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        update: [null, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```
Comment (czoido) [inline conan/internal/default_settings.py]:
```suggestion
        update: [null, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

File: conan/internal/api/detect/detect_vs.py [hunk 2]
```diff
@@ -14,8 +14,11 @@ def vs_detect_update(version):
     version = {"194": "17", "193": "17", "192": "16", "191": "15"}.get(str(version))
     full_version = _vs_installation_path(version)[1]
     components = full_version.split(".")
+    # 17.10 --> 1940
+    # 17.11 --> 1941
+    # etc...    
     if len(components) > 1:
-        return components[1]
+        return str(int(components[1]) % 10)
```
Comment (czoido) [inline conan/internal/api/detect/detect_vs.py]: We have 17.11 in Github Actions and it returned 11 as update

File: .github/workflows/win-tests.yml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  testing:
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ['3.8', '3.6']
```
Comment (czoido) [inline .github/workflows/win-tests.yml]:
```suggestion
        python-version: ['3.13', '3.8', '3.6']
```

File: test/functional/test_profile_detect_api.py [hunk 4]
```diff
@@ -21,15 +21,17 @@ def test_profile_detect_compiler(self):
             compiler.version={{detect_api.default_compiler_version(compiler, version)}}
             compiler.runtime={{runtime}}
             compiler.cppstd={{detect_api.default_cppstd(compiler, version)}}
-            compiler.update={{detect_api.detect_msvc_update(version)}}
+            # FIXME: check if we can do something better than this, we have a problem with updates
+            compiler.update={{ (detect_api.detect_msvc_update(version) | int) % 10 }}
```
Comment (memsharded) [inline test/functional/test_profile_detect_api.py:26]: Recall to update the docs with this

File: test/functional/toolchains/microsoft/test_v2_msbuild_template.py [hunk 5]
```diff
@@ -49,11 +49,11 @@ def test_msbuild_lib_2022():
     assert "hello/0.1: _MSC_VER191" in client.out
 
     # Create works
-    client.run("create . -s compiler.version=193")
+    client.run("create . -s compiler.version=194")
```
Comment (memsharded) [inline test/functional/toolchains/microsoft/test_v2_msbuild_template.py]: I am not sure why this change.
previously neither 193 nor 194 were the default version, yet the test managed to build and pass. Why is it necessary to do this, maybe because 193 toolset is not installed anymore? The problem with this is that we would not test anymore that we can build ``compiler.version=193`` with a VS>=17.10

File: test/integration/toolchains/microsoft/vcvars_test.py [hunk 6]
```diff
@@ -137,10 +138,10 @@ class TestConan(ConanFile):
     client.save({"conanfile.py": conanfile})
     client.run('install . -s os=Windows -s compiler=msvc -s compiler.version=193 '
                '-s compiler.cppstd=14 -s compiler.runtime=static '
-               '-c tools.microsoft:winsdk_version=8.1')
+               '-c tools.microsoft:winsdk_version=17.11')
```
Comment (memsharded) [inline test/integration/toolchains/microsoft/vcvars_test.py]: Is this a valid SDK version? I'd say it is not.

File: test/unittests/util/output_test.py [hunk 7]
```diff
@@ -54,23 +54,24 @@ def test_unzip_output(self):
         content = load(os.path.join(output_dir, "example.txt"))
         self.assertEqual(content, "Hello world!")
 
-    @pytest.mark.skipif(platform.system() != "Windows", reason="Requires windows")
-    def test_short_paths_unzip_output(self):
-        tmp_dir = temp_folder()
-        file_path = os.path.join(tmp_dir, "src/"*40, "example.txt")
-        save(file_path, "Hello world!")
+    # FIXME: revisit this test
+    # @pytest.mark.skipif(platform.system() != "Windows", reason="Requires windows")
```
Comment (memsharded) [inline test/unittests/util/output_test.py]: Mark as "skip" instead of commented

File: test/integration/toolchains/gnu/test_makedeps.py [hunk 8]
```diff
@@ -411,6 +411,7 @@ def package_info(self):
     assert "CONAN_NAME_OTHER" not in make_content
 
 
+@pytest.mark.skipif(platform.system() == "Windows", reason="Test uses Unix-style paths not compatible with Windows filesystem.")
```
Comment (memsharded) [inline test/integration/toolchains/gnu/test_makedeps.py]: This is unexpected, at least some ``#FIXME`` (as it is not an xfail), better investigate a bit more?

File: test/functional/toolchains/test_nmake_toolchain.py [hunk 9]
```diff
@@ -11,11 +11,11 @@
...
     "compiler, version, runtime, cppstd, build_type, defines, cflags, cxxflags, sharedlinkflags, exelinkflags",
     [
-        ("msvc", "190", "dynamic", "14", "Release", [], [], [], [], []),
-        ("msvc", "190", "dynamic", "14", "Release",
-         ["TEST_DEFINITION1", "TEST_DEFINITION2=0", "TEST_DEFINITION3=", "TEST_DEFINITION4=TestPpdValue4",
-          "TEST_DEFINITION5=__declspec(dllexport)", "TEST_DEFINITION6=foo bar"],
-         ["/GL"], ["/GL"], ["/LTCG"], ["/LTCG"]),
+        # ("msvc", "190", "dynamic", "14", "Release", [], [], [], [], []),
+        # ("msvc", "190", "dynamic", "14", "Release",
+        #  ["TEST_DEFINITION1", "TEST_DEFINITION2=0", "TEST_DEFINITION3=", "TEST_DEFINITION4=TestPpdValue4",
```
Comment (memsharded) [inline test/functional/toolchains/test_nmake_toolchain.py]: At least a  ``#FIXME``, to recover this, even if 190 is not available, at least to test the defines and flags

**Review by memsharded — COMMENTED** [review]
This is big. Great work.
Just some minor concerns.

File: test/functional/toolchains/microsoft/test_v2_msbuild_template.py (hunk 5 above)
Comment (czoido) [reply test/functional/toolchains/microsoft/test_v2_msbuild_template.py]: ok, let me try to install the v143 toolset too and check if we can make it work as it should.
